go/integTest/cluster.go

```diff
@@ -0,0 +1,221 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package integTest
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/valkey-io/valkey-glide/go/v2/config"
+)
+
+// ValkeyCluster represents a test cluster instance
+type ValkeyCluster struct {
+	ClusterFolder string
+	NodesAddr     []config.NodeAddress
+	TLS           bool
+}
+
+// NewValkeyCluster creates a new ValkeyCluster instance
+func NewValkeyCluster(tls bool, opts ...ClusterOption) (*ValkeyCluster, error) {
```

Comment:
please re-use code to start/stop cluster from `go/integTest/glide_test_suite_test.go`